ionc/ion_binary.h

```diff
@@ -91,7 +93,12 @@ ION_API_EXPORT int ion_binary_len_int_64_unsigned(uint64_t value);
 /** Get the size of ion binary representation fields of the given double value. Fixed at sizeof(double)
  *
  */
-ION_API_EXPORT int ion_binary_len_ion_float(double value);
+ION_API_EXPORT int ion_binary_len_ion_double(double value);
+
+/** Get the size of ion binary representation fields of the given float value. Fixed at sizeof(float)
+ *
+ */
+ION_API_EXPORT int ion_binary_len_ion_float(float value);
```

Comment:
Unfortunately we can't change the signature of the existing function because that could break any user of `ion_binary_len_ion_float` that provided it a full-length `double`.

Let's leave the existing function signatures as-is and add new ones with a `float_32` naming scheme.

So we'd have

```c
ION_API_EXPORT int ion_binary_len_ion_float(double value);
ION_API_EXPORT int ion_binary_len_ion_float_32(float value);
```

Reply:
Yeah I should have probably asked you about this one! Well it makes sense to have a float_32 rather than float and double explicitly.